README.md

```diff
@@ -1 +1,161 @@
+      30 -- time_extend       -- in case if time difference more than 27 bits, we use that to handle time difference*
+      31 -- time_stamp        -- probably just time_stamp? =(
+      
+      * `array[1] = (time_delta & TIME_DELTA_MASK) >> TS_SHIFT`, 
+         this one is only one actually useful and well described case
+         (even in `trace-cmd` code, so other cases probably **dead**)
+   ```
+   
++ FTRACE EVENT FORMATS - contains an events format descriptions, also could be found here: `debugfs/tracing/events/ftrace/<event>/format`
+
+   We need `FUNCTION` event, here is an example (debug output of `sel4-ftrace-merge` tool):
+   ```[DBG]:
+   [DBG]: ------------- Ftrace Event Formats -------------
+   [DBG]: Ftrace event formats count: 15
+   .....
+   [DBG]: Event Format #8 size: 0x1cc
+   [DBG]: Event Format:
+   name: function
+   ID: 1
+   format:
+           field:unsigned short common_type;       offset:0;       size:2; signed:0;
+           field:unsigned char common_flags;       offset:2;       size:1; signed:0;
+           field:unsigned char common_preempt_count;       offset:3;       size:1; signed:0;
+           field:int common_pid;   offset:4;       size:4; signed:1;
+
+           field:unsigned long ip; offset:8;       size:8; signed:0;
+           field:unsigned long parent_ip;  offset:16;      size:8; signed:0;
+
+   print fmt: " %ps <-- %ps", (void *)REC->ip, (void *)REC->parent_ip
+   ```
+   Some format description:
+   ```
+      common_type          -- should be equal to the function event ID ( ID: 1 here)
+      common_flags         -- some flags
+      common_preempt_count -- don't know what is it for =(
+      common_pid           -- linux process ID, for seL4 we use here some dedicated PID (like: 999, also hardcoded =)
+      ip                   -- function IP, for for seL4 we use here some dedicated IP ( also hardcoded =)
+      parent_ip            -- parent function IP (caller), for for seL4 we use here some dedicated IP ( also hardcoded =)
+   ```
+   
++ PAGE FORMAT example:
+   ```
+   PAGE_HEADER {
+      ...
+      data {
+        EVENT_HEADER {
+           ...
+           array {
+              EVENT_FORMAT {}
+           }
+        },
+        EVENT_HEADER {
+           ...
+           array {
+              EVENT_FORMAT {}
+           }
+        },
+        EVENT_HEADER {
+           ...
+           array {
+              EVENT_FORMAT {}
+           }
+        },
+        ...
+      }
+   }
+   ```
+
++ PROCESS INFORMATION   -- section contains current processes information from: `debugfs/tracing/saved_cmdlines`
+
+   Here we need to add our seL4 `common_pid` and it's name
+
++ KALLSYMS INFORMATION  -- section contains current kernel address to function maps from: `/proc/kallsyms`
+
+   Here we need to add our seL4 `ip`, `parent_ip` and their names
+
+### Useful links
+
+[man trace-cmd](https://man7.org/linux/man-pages/man5/trace-cmd.dat.5.html)  
```

Comment:
As mentioned above, some of these could be reference style links, and some could be like they currently are. Your call :)

Reply:
Sure, thank you for all your suggestions. But here I think that there is not much difference to use this approach or any other one. So keep them as is.

Reply:
Alright, np